wagtail/core/models.py

```diff
@@ -845,6 +1165,23 @@ def cached_content_type(self):
         """
         return ContentType.objects.get_for_id(self.content_type_id)
 
+    @cached_property
```

Comment:
Again, caching this could be slightly risky if the active locale changes over the lifetime of the object - caching the result of get_translation would avoid this risk.